example-dotnet/Program.cs

```diff
@@ -11,8 +22,134 @@ static void Main(string[] args)
             Say.hello("example");
             Console.WriteLine("Hello World!");
 
+	    // Usage of BCrypt.Net API
+
             var hashedPassword = BCrypt.Net.BCrypt.HashPassword("my-secret-password");
             BCrypt.Net.BCrypt.Verify("wrong-password", hashedPassword);
+
+            // Usage of SharpZipLib API
+
+	    // This can trigger an exploit, see:
+	    // https://github.com/icsharpcode/SharpZipLib/issues/232
+	    // The usage of SharpZipLib API here will be reported as
+	    // may be reaching one of the methods affected by an
+	    // update from vulnerable version to the safe version.
+
+	    // `FastZip().ExtractZip()` from SharpZipLib 1.0.0-alpha1
+	    // is known to be vulnerable (CVE-2018-1002208), see also:
+	    // https://github.com/icsharpcode/SharpZipLib/issues/232. Latest
+	    // version 1.1.0 without the vulnerability has a
+	    // compatible API. The software can be updated to
+	    // 1.0.0-rc1 to resolve the issue. 1.0.0-alpha1 includes
+	    // netstandard1.3 framework version, while 1.0.0-rc1
+	    // includes net45 and netstandard2.0 framework
+	    // versions. The usage of the vulnerable API is commented
+	    // out, and we instead opted to include a simple access to
+	    // a constant in the library. This is to exemplify the
+	    // negative case of breaking update over different sets of
+	    // frameworks between the current version and the
+	    // recommended update.
+
+            // Console.WriteLine("Unzipping example ZIP file using SharpZipLib");
+            // new FastZip().ExtractZip(@"resources" + Path.DirectorySeparatorChar + "example.zip", @".", null);
+
+            // Here we simply retrieve a constant within SharpZipLib
+            // to avoid method call.
+            Console.WriteLine("SharpZipLib ZipConstants.VersionMadeBy = " + ZipConstants.VersionMadeBy);
+
+            // Usage of DotNetZip API
+      
+            // Unzip the file to memory using DotNetZip
+            // (vulnerable). This may be reaching methods affected by
+            // the update. The zip extraction API of DotNetZip 1.10.0

```

Comment:
> **may be** reaching methods affected by the update

Guessing the uncertainty comes from our use of "Maybe" in reports. I'm considering changing the wording to "Yes" (in the sense that our call graph analysis reports a "Yes", ignoring false positives).

Reply:
I'm still using "Maybe" though, but I put those in the tables, which hopefully explains everything in a clearer way.